src/test/java/tr/com/srdc/cda2fhir/AllergyConcernActTest.java

```diff
@@ -0,0 +1,391 @@
+		entryRelationship2.setObservation(reactionObservation2);
+		observation.getEntryRelationships().add(entryRelationship2);
+		
+		DiagnosticChain dxChain = new BasicDiagnostic();
+		Boolean validation = act.validateAllergyProblemActAllergyObservation(dxChain, null);
+		Assert.assertTrue(validation);		
+
+		Bundle bundle = rt.tAllergyProblemAct2AllergyIntolerance(act);
+		AllergyIntolerance allergyIntolerance = findOneResource(bundle);
+		List<AllergyIntoleranceReactionComponent> reactions = allergyIntolerance.getReaction();
+		
+		AllergyIntoleranceReactionComponent reaction1 = reactions.get(0);
+		AllergyIntoleranceReactionComponent reaction2 = reactions.get(1);
+		
+		String actual1 = reaction1.getOnsetElement().getValueAsString();
+		String actual2 = reaction2.getOnsetElement().getValueAsString();
+	
+		Assert.assertEquals(expected1, actual1.replaceAll("-", ""));	
+		Assert.assertEquals(expected2, actual2.replaceAll("-", ""));
+		
+		String lastActual = allergyIntolerance.getLastOccurrenceElement().getValueAsString();
+		Assert.assertEquals(expected2, lastActual.replaceAll("-", ""));
+	}
+
+	@Test
+	public void testEffectiveTime() throws Exception {
+		AllergyProblemActImpl act = createAllergyConcernAct();
+		DiagnosticChain dxChain = new BasicDiagnostic();
+
+		String expected1 = "20171002";		
+		IVL_TS ivlTs1 = createEffectiveTimeLow(expected1);
+		act.setEffectiveTime(ivlTs1);
+		
+		Boolean validation1 = act.validateAllergyProblemActEffectiveTime(dxChain, null);
+		Assert.assertTrue(validation1);
+
+		Bundle bundle1 = rt.tAllergyProblemAct2AllergyIntolerance(act);
+		AllergyIntolerance allergyIntolerance1 = findOneResource(bundle1);
+		String actual1 = allergyIntolerance1.getAssertedDateElement().getValueAsString();
+		Assert.assertEquals(expected1, actual1.replaceAll("-", ""));
+
+		String expected2 = "20161103";
+		IVL_TS ivlTs2 = createEffectiveTimeValue(expected2);
+		act.setEffectiveTime(ivlTs2);
+
+		Boolean validation2 = act.validateAllergyProblemActEffectiveTime(dxChain, null);
+		Assert.assertTrue(validation2);
+				
+		Bundle bundle2 = rt.tAllergyProblemAct2AllergyIntolerance(act);
+		AllergyIntolerance allergyIntolerance2 = findOneResource(bundle2);
+		String actual2 = allergyIntolerance2.getAssertedDateElement().getValueAsString();
+		Assert.assertEquals(expected2, actual2.replaceAll("-", ""));	
+	}
+		
+	static private void verifyAllergyIntoleranceVerificationStatus(AllergyProblemAct act, String expected) throws Exception {
+		Bundle bundle = rt.tAllergyProblemAct2AllergyIntolerance(act);
+		AllergyIntolerance allergyIntolerance = findOneResource(bundle);
+		
+    	AllergyIntoleranceVerificationStatus verificationStatus = allergyIntolerance.getVerificationStatus();
+    	String actual = verificationStatus == null ? null : verificationStatus.toCode();
+		Assert.assertEquals(expected, actual);		
+	}
+	
+	@Test
+	public void testAllergyIntoleranceStatusCode() throws Exception {
```

Comment:
would you mind adding a test/assertion here for statusCode NULL. Just want to make sure we aren't defaulting anything to active when nothing arrives on the record. It isn't required per the spec, and doesn't seem to be on any of the test CCDAs that come with the lib.

Reply:
there was one and added another one after specifically setting status code to zero.